packages/react-portal/src/components/Portal/usePortal.ts

```diff
@@ -14,10 +16,17 @@ const mergeProps = makeMergeProps<PortalState>();
  * @param defaultProps - (optional) default prop values provided by the implementing type
  */
 export const usePortal = (props: PortalProps, defaultProps?: PortalProps): PortalState => {
-  const state = mergeProps({ shouldRender: !useIsSSR() }, defaultProps, props);
+  const virtualParentRootRef = React.useRef<HTMLSpanElement>(null);
+  const state = mergeProps({ shouldRender: !useIsSSR(), virtualParentRootRef }, defaultProps, props);
   const fallbackMountNode = usePortalMountNode({ disabled: !!state.mountNode });
 
   state.mountNode = state.mountNode ?? fallbackMountNode;
 
+  useIsomorphicLayoutEffect(() => {
```

Comment:
```suggestion
  React.useEffect(() => {
```

We can use there `useEffect()` as any interactions that could call `elementContains()` will occur _after_ DOM painting.